src/js/utils/DSLParserUtil.js

```diff
@@ -0,0 +1,196 @@
+ * The `filters` object is a library with the available filters that can be used
+ * in this context, while the `resultset` is an instance of `List` or `Tree` on
+ * which the filers have to be applied.
+ *
+ * @private
+ * @param {CombineNode} ast - The AST node for the combine operation
+ * @param {function} filter1 - The first child filter function (auto-bound)
+ * @param {function} filter2 - The second child filter function (auto-bound)
+ * @returns {function} - Returns the fabricated combined function
+ */
+function combineFiltersFactory(ast, filter1, filter2) {
+  switch (ast.combinerType) {
+    case DSLCombinerTypes.AND:
+      return function (filters, resultset) {
+        // We are interested in the intersection of the results of the two filters
+        let intermediateResultset = filter1(filters, resultset);
+        return filter2(filters, intermediateResultset);
+      };
+
+    case DSLCombinerTypes.OR:
+      return function (filters, resultset) {
+        // We are interested in the union of the results of the two filters
+
+        // TODO: Merge the results of `filter1(filters, resultset)` and
+        //       `filter2(filters, resultset)`.
+
+        // TODO: Implement a `List.merge` function that merges two lists,
+        //       discarding duplicate values.
+
+        return resultset;
+      };
+  };
+}
+
+/**
+ * Template function for filtering a resultset, used by the parser
+ *
+ * When this function is 'instantiated' by the parser, `this` context
+ * is bound to an instance of a `FilterNode` representing the token AST node.
+ *
+ * When this function is used, the user has to supply the final two arguments.
+ * The `filters` object is a library with the available filters that can be used
+ * in this context, while the `resultset` is an instance of `List` or `Tree` on
+ * which the filers have to be applied.
+ *
+ * @private
+ * @param {FilterNode} ast - The AST node for the filter operation
+ * @param {Object} filters - An object containing the valid filters that can be used
+ * @param {List} resultset - An instance of List or Tree containing the items to filter
+ *
+ * @returns {function} - Returns the fabricated filter function
+ */
+/* eslint-disable no-unused-vars */
+function filterFunctionFactory(ast) {
+  return function (filters, resultset) {
+    // TODO: Lookup in `filters` object one or more valid filters that can be
+    //       applied on the bound token. We can use the `this.filterType` and
+    //       `this.filterParams` for this purpose. Then apply them on the
+    //       resultset, keeping only the matched results.
+
+    return resultset;
+  };
+};
+/* eslint-enable no-unused-vars */
+
+/**
+ * The following functions are used by the JISON parser in order to parse
+ * the expression into a properly nested set of functions and AST nodes.
+ *
+ * @name DSLParserUtil
+ */
+module.exports = {
+
+  /**
+   * Namespace for the merge operator
```

Comment:
A compromise between camel-case, snake-case and other weird function naming (see discussion above). Makes readability of the `jison` document slightly better.